arangod/Aql/Variable.h

```diff
@@ -76,6 +85,18 @@ struct Variable {
 
   bool isEqualTo(Variable const& other) const;
   
+  /// @brief returns the type of the variable. The type is determined based
+  // on the constantValue. If constantValue.isNone, the type is Type::Regulor,
+  // otherwise it is Type::Const
+  Type type() const noexcept;
+
+  /// @brief returns the constant value of the variable.
+  AqlValue constantValue() const noexcept { return _constantValue; }
+
+  /// @brief set the constant value of the variable.
+  /// This implicitly changes the type -> see type()
+  void setConstantValue(AqlValue value) noexcept;
```

Comment:
With that, we will have `constValue()` and `constantValue()` which is confusing from an API perspective.
Can we rename the older `constValue()` function to something completely different here, so the two are not confused? Mid-term we should also clean this up, but a short-term fix could be just renaming the functions.